feat: New Pattern Module avm/ptn/sa/document-knowledge-mining

Prajwal-Microsoft opened the pull request:
## Description
This module contains the resources required to deploy the [Document knowledge mining solution accelerator](https://github.com/microsoft/Document-Knowledge-Mining-Solution-Accelerator) for both Sandbox environments and WAF aligned environments.
<!--
>Thank you for your contribution !
> Please include a summary of the change and which issue is fixed.
> Please also include the context.
> List any dependencies that are required for this change.

Fixes #123
Closes #123
-->

## Pipeline Reference

<!-- Insert your Pipeline Status Badge below -->

| Pipeline |
| -------- |
[![avm.ptn.sa.document-knowledge-mining](https://github.com/Ravikirana-Microsoft/bicep-registry-modules/actions/workflows/avm.ptn.sa.document-knowledge-mining.yml/badge.svg)](https://github.com/Ravikirana-Microsoft/bicep-registry-modules/actions/workflows/avm.ptn.sa.document-knowledge-mining.yml)

## Type of Change

<!-- Use the checkboxes [x] on the options that are relevant. -->

- Azure Verified Module updates:
  - [ ] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation
- [ ] Update to CI Environment or utilities (Non-module affecting changes)

## Checklist

- [ ] I'm sure there are no other open Pull Requests for the same update/change
- [ ] I have run `Set-AVMModule` locally to generate the supporting module files.
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] I have updated the module's CHANGELOG.md file with an entry for the next version

<!--  Please keep up to date with the contribution guide at https://aka.ms/avm/contribute/bicep -->

AlexanderSehr commented on avm/ptn/sa/document-knowledge-mining/main.bicep:
```diff
@@ -0,0 +1,1087 @@
+}
+
+// ========== Container Registry ========== //
+module avmContainerRegistry './modules/container-registry.bicep' = {
```
Just curious, why was this moved into a nested module if the scope does not change?

AlexanderSehr reviewed (commented):
Great work @Prajwal-Microsoft, just a few comments 💪

Prajwal-Microsoft commented on avm/ptn/sa/document-knowledge-mining/tests/e2e/sandbox/main.test.bicep:
```diff
@@ -0,0 +1,62 @@
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+// e.g., for a module 'network/private-endpoint' you could use 'dep-dev-network.privateendpoints-${serviceShort}-rg'
```
Remove these comments.

Prajwal-Microsoft commented:
> Great work @Prajwal-Microsoft, just a few comments 💪

Thank you @AlexanderSehr  for reviewing and providing the feedback.

I have addressed the feedback. Reques you to review the changes.
